app/models/list_import/anime_planet/row.rb

```diff
@@ -0,0 +1,159 @@
+class ListImport
+  class AnimePlanet
+    class Row
+      attr_reader :node, :type
+
+      def initialize(node, type)
+        @node = node
+        @type = type
+      end
+
+      def media
+        key = "#{type}/#{media_info[:id]}"
+
+        Mapping.lookup('animeplanet', key) ||
+          Mapping.guess(type.classify.safe_constantize, media_info)
+      end
+
+      def media_info
+        {
+          id: node.attr('data-id')&.to_i,
+          title: media_title(tooltip),
+          show_type: show_type(tooltip),
+          episode_count: total_episodes(tooltip),
+          chapter_count: total_chapters(tooltip)
+        }.compact
+      end
+
+      def status
+        @status ||= media_status(tooltip)
+
+        case @status
+        when /ing\z/ then :current
+        when /\AWant/ then :planned
+        when /\AWon't/ then nil
+        when 'Stalled' then :on_hold
+        when 'Dropped' then :dropped
+        when 'Watched', /Read([0-9.]+)?/ then :completed
+        end
+      end
+
+      def progress
+        if status == :completed
+          if type == 'anime'
+            media_info[:episode_count]
+          else
+            media_info[:chapter_count]
+          end
+        else
+          media_progress(tooltip)
+        end
+      end
+
+      def rating
+        tooltip.css('.ttRating')&.last&.text&.to_f
+      end
+
+      def reconsume_count
+        return unless type == 'anime' && status == :completed
+
+        amount = tooltip.at_css('.myListBar')&.content
+        amount&.split('-')&.last&.split('x')&.first&.to_i
+      end
+
+      def volumes
+        return unless type == 'manga'
+
+        if status == :completed
+          total_volumes(tooltip)
+        else
+          read_volumes(tooltip)
+        end
+      end
+
+      def data
+        %i[status progress rating reconsume_count volumes].map { |k|
+          [k, send(k)]
+        }.to_h
+      end
+
+      private
+
+      def tooltip
+        @tooltip ||= Nokogiri::HTML.fragment(
+          node.at_css('.tooltip').attr('title')
+        )
+      end
+
+      # Media Info
+      def media_title(fragment)
```

Comment:
Why not just access `tooltip` directly within the method and not take an argument?